integrationtests/cli/apply/apply_test.go

```diff
@@ -237,3 +244,268 @@ var _ = Describe("Fleet apply", Ordered, func() {
 		})
 	})
 })
+
+var _ = Describe("Fleet apply with helm charts with dependencies", Ordered, func() {
+
+	var (
+		dirs      []string
+		name      string
+		options   apply.Options
+		tmpDirRel string
+		tmpDir    string
+		repo      = repository{
+			port: port,
+		}
+	)
+
+	JustBeforeEach(func() {
+		// start a fake helm repository
+		repo.startRepository(false)
+		tmpDir = GinkgoT().TempDir()
+		err := cp.Copy(path.Join(cli.AssetsPath, "deps-charts", name), tmpDir)
+		Expect(err).NotTo(HaveOccurred())
+		// get the relative path because fleet apply needs a relative path
+		pwd, err := os.Getwd()
+		Expect(err).NotTo(HaveOccurred())
+		tmpDirRel, err = filepath.Rel(pwd, tmpDir)
+		Expect(err).NotTo(HaveOccurred())
+		dirs = []string{tmpDirRel}
+		err = fleetApply(name, dirs, options)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	When("folder contains helm chart with no fleet.yaml", func() {
+		BeforeEach(func() {
+			name = "no-fleet-yaml"
+		})
+
+		It("creates a Bundle  with all the resources, including the dependencies", func() {
+			Eventually(func() bool {
+				bundle, err := cli.GetBundleFromOutput(buf)
+				Expect(err).NotTo(HaveOccurred())
+				// files expected are:
+				// Chart.yaml + values.yaml + templates/configmap.yaml +
+				// Chart.lock + charts/config-chart-0.1.0.tgz
+				Expect(len(bundle.Spec.Resources)).To(Equal(5))
+				files, err := getAllFilesInDir(tmpDir)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(files)).To(Equal(len(bundle.Spec.Resources)))
+				for _, file := range files {
+					presentInBundleResources(file, bundle.Spec.Resources)
+				}
+				// explicitly check for dependency files
+				presentInBundleResources(path.Join(tmpDir, "Chart.lock"), bundle.Spec.Resources)
+				presentInBundleResources(path.Join(tmpDir, "charts/config-chart-0.1.0.tgz"), bundle.Spec.Resources)
+				return true
+			}).Should(BeTrue())
```

Comment:
I think this doesn't make much sense: either we should return an expression, which may or may not evaluate as `true`, or skip this altogether: do we even _need_ `Eventually` blocks here?

Reply:
Since the test uses the cli to create the bundle it will be ready when the cli exists. The buffer `buf` will be in it's final state, as soon as `cli.GetBundleFromOutput` returns. `Eventually` is great when we're waiting for resources to appear in k8s. If I understood that correctly, I'd definitely remove them.

I also see a lot of helper functions like `onlyPresentInBundleResources`, these could be custom matchers: https://onsi.github.io/gomega/#adding-your-own-matchers 
But I don't think we're using those anywhere else, no need to do change it.

Reply:
I've tested deleting the `Eventually` statements and running in a loop for 10000 times and it passed.
So... `Eventually` statements deleted. 

I've also added a couple of custom matchers and deleted the helpers.